src/cp_test.ts

```diff
@@ -1,28 +1,55 @@
 import { anything, anyFunction, instance, mock, verify, when } from 'ts-mockito';
 import * as querystring from 'querystring';
+import { expect } from 'chai';
 import WebSocket = require('isomorphic-ws');
-
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import { tmpdir } from 'os';
+import * as tar from 'tar';
 import { CallAwaiter } from '../test';
 import { KubeConfig } from './config';
 import { Exec } from './exec';
 import { Cp } from './cp';
-import { WebSocketHandler, WebSocketInterface } from './web-socket-handler';
+import { BinaryHandler, WebSocketHandler, WebSocketInterface } from './web-socket-handler';
+import { V1Status } from './api';
+import { randomUUID } from 'crypto';
+import { sleep } from './util';
 
 describe('Cp', () => {
+    let tmpDir: string | undefined;
```

Comment:
If you initialize `tmpDir` here with an empty string you don't need to check in every test for it, you could even initialize it with a proper dir:

```suggestion
    let tmpDir: string = `${tmpdir()}/${randomUUID()}`;
```